assets/src/editor/events/dates-and-times/data/with-ticket-entities-for-all-date-entities.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External imports
+ */
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal imports
+ */
+import { getDateEntityIds } from './utils';
+
+/**
+ * withTicketEntitiesRelatedToAllDateEntities
+ * returns an object containing the following:
+ * 	ticketEntities - an array of ticket entities for the supplied event date
+ * 	ticketEntitiesLoaded - boolean true if loading is complete
+ *
+ * @return {function}
+ */
+export const withTicketEntitiesForAllDateEntities = createHigherOrderComponent(
+	withSelect(
+		( select, { dateEntities, dateEntitiesLoaded } ) => {
+			if ( dateEntitiesLoaded ) {
+				const { getRelatedEntitiesForIds } = select( 'eventespresso/core' );
+				const { hasFinishedResolution } = select( 'core/data' );
+				const dateIds = getDateEntityIds( dateEntities );
```

Comment:
Generally, I'm not as concerned about internally scoped variable names because the context helps with clarity.  I mostly thought of scenarios where I might see a prop passed through or a component name implemented where the additional clarity (and consistency) helps.